Support torch randint ops.

fukatani opened the pull request:
Resolve #1579

fukatani commented:
BTW, I'm participated in [Hacktoberfest 2022](https://hacktoberfest.com/participation/).
I appreciate if you could add hacktoberfest-accepted to this PR and https://github.com/apple/coremltools/pull/1630.
(like [here](https://github.com/Coder-forfun/Hactoberfest-accepted/pull/528))

thanks!

TobyRoseman commented on coremltools/converters/mil/frontend/torch/test/test_torch_ops.py:
```diff
@@ -2366,6 +2368,26 @@ def forward(self, x):
+    def test_randint(self, backend, shape, low, high):
+        class Randint(nn.Module):
+            def forward(self, x):
+                return torch.randint(low, high, x.shape)
```
Something we have done before in our unit tests when the model is nondeterministic [is to just return the length](https://github.com/apple/coremltools/blob/f7fba76f15295719c7ec9e5dddefcd3973b74c03/coremltools/converters/mil/frontend/torch/test/test_torch_ops.py#L1488). This allows us to use our existing testing helper functions.

TobyRoseman commented on coremltools/converters/mil/frontend/torch/test/test_torch_ops.py:
```diff
@@ -2366,6 +2368,26 @@ def forward(self, x):
+
+        torch_in = torch.tensor(shape, dtype=torch.int32)
+        model = Randint().eval()
+        verify = lambda x: numpy.testing.assert_array_less(x, high) and numpy.testing.assert_array_less(low, x)
```
I do like checking that `low < x < high`. However I don't think it justifies adding a lot more testing functionality.

If you want, you can do the model conversion yourself in this function. Then get the output and do this check.

It's also fine if you just want to modify the Torch model to return the length. Then use our existing testing helper functions.

TobyRoseman commented:
> I appreciate if you could add hacktoberfest-accepted to this PR and #1630. (like [here](https://github.com/Coder-forfun/Hactoberfest-accepted/pull/528))

I'm not clear what you are asking me to do. Are you asking me to create a GitHub label in this repository called "hacktoberfest-accepted"?

fukatani commented:
@TobyRoseman 
Thanks for your kind review. I addressed review comments.

> I'm not clear what you are asking me to do. Are you asking me to create a GitHub label in this repository called "hacktoberfest-accepted"?

Sorry for lack of explanation.

There are two methods and either is enough.
- [Add "hacktoberfest-accepted" label to **PR**.](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels)
- [Add "hacktoberfest" topics to **repository**.](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/classifying-your-repository-with-topics)

If you do latter, all PRs to this repo can be recognized by hactoberfest.

TobyRoseman commented:
Thanks for addressing my feedback. This change looks good. I have kicked off a [CI run](https://gitlab.com/coremltools1/coremltools/-/pipelines/665468028). Assuming that run is good, I will merge this change.

TobyRoseman commented:
CI run is good. Thanks @fukatani!
